x/ccv/provider/keeper/hooks.go

```diff
@@ -175,7 +175,61 @@ func (h Hooks) AfterProposalVotingPeriodEnded(ctx sdk.Context, proposalID uint64
 func (h Hooks) AfterProposalDeposit(ctx sdk.Context, proposalID uint64, depositorAddr sdk.AccAddress) {
 }
 
+// AfterProposalVote opts in validators that vote YES (with 100% weight) on a `ConsumerAdditionProposal`. If a
+// validator votes multiple times, only the last vote would be considered on whether the validator is opted in or not.
 func (h Hooks) AfterProposalVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress) {
+	validator, found := h.k.stakingKeeper.GetValidator(ctx, voterAddr.Bytes())
+	if !found {
+		return
+	}
+
+	consAddr, err := validator.GetConsAddr()
+	if err != nil {
+		h.k.Logger(ctx).Error("could not extract validator's consensus address",
+			"error", err.Error(),
+			"validator acc addr", voterAddr,
+		)
+		return
+	}
+
+	chainID, found := h.k.GetProposedConsumerChain(ctx, proposalID)
+	if !found {
+		return
+	}
+
+	vote, found := h.k.govKeeper.GetVote(ctx, proposalID, voterAddr)
+	if !found {
+		h.k.Logger(ctx).Error("could not find vote for validator",
+			"validator acc addr", voterAddr,
+			"proposalID", proposalID,
+		)
+		return
+	}
+
+	if len(vote.Options) == 1 && vote.Options[0].Option == v1.VoteOption_VOTE_OPTION_YES {
```

Comment:
Need to be careful here, appearantly it *might* be that votes are created that use the vote.Option field instead of the vote.Options field, see 

```
// Deprecated: Prefer to use `options` instead. This field is set in queries
  // if and only if `len(options) == 1` and that option has weight 1. In all
  // other cases, this field will default to VOTE_OPTION_UNSPECIFIED.
  VoteOption option = 3 [deprecated = true];
```

from https://docs.cosmos.network/v0.47/build/modules/gov#vote

maybe also check for this just in case, it's deprecated but not removed for a reason I guess

Reply:
Thanks for pointing this out.
The Governance keeper exposes:
```
GetVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress) (vote v1.Vote, found bool)
```
that I use to retrieve a `vote`. There's no `vote.Option` in `v1.Vote`. 
`vote.Option` appears in legacy `v1beta1.MsgVote` that can be used in [`Vote`](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/gov/keeper/msg_server.go#L244). But `Vote` [internally](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/gov/keeper/msg_server.go#L120) would build the `Options` field. So, when we extract the vote, it just has `Options`.

Reply:
Thanks for double-checking!